### ISSUE TITLE
Wrong reconciler interval for NATS backend (kyma 1.24.8)

### ISSUE THREAD

**Issue by radufa** [kyma-project/kyma#12979]
**Description**

Description
If a subscription is not changed (touched) a reconciliation call is triggered each 10min instead of 10h (the default value).
Cause on NATS backend each reconciliation call deletes the associated NATS subject, the messages sent to this subject are lost.
So that a lot of messages are lost on a production system, under load.
It seems that the reconciliation interval used for BEB beckend is wrongly used for NATS too...

**Expected result**

The reconciliation call should be triggered only each 10h (the default value).

**Actual result**
The reconciliation call is triggered each 10mins for the only one ready subscription "test" (in the namespace "test").
Extract from controller log file:
```
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:135","message":"received subscription reconciliation request","context":{"namespace":"test","name":"test"}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"handlers/nats.go:129","message":"connection status","context":{"status":1}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"handlers/nats.go:145","message":"successfully unsubscribed/deleted subscription","context":{}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"eventtype/clean.go:45","message":"cannot find application","context":{"name":"noapp"}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"eventtype/clean.go:53","message":"clean event-type","context":{"before":"sap.kyma.custom.noapp.order.created.v1","after":"sap.kyma.custom.noapp.order.created.v1"}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:221","message":"successfully created Nats subscriptions","context":{"kind":"Subscription","name":"test","namespace":"test","version":2}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:266","message":"successfully updated subscription status","context":{}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:135","message":"received subscription reconciliation request","context":{"namespace":"test","name":"test"}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"handlers/nats.go:129","message":"connection status","context":{"status":1}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"handlers/nats.go:145","message":"successfully unsubscribed/deleted subscription","context":{}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"eventtype/clean.go:45","message":"cannot find application","context":{"name":"noapp"}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"eventtype/clean.go:53","message":"clean event-type","context":{"before":"sap.kyma.custom.noapp.order.created.v1","after":"sap.kyma.custom.noapp.order.created.v1"}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:221","message":"successfully created Nats subscriptions","context":{"kind":"Subscription","name":"test","namespace":"test","version":2}}
{"level":"INFO","timestamp":"2022-01-04T08:23:36Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:266","message":"successfully updated subscription status","context":{}}


{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:135","message":"received subscription reconciliation request","context":{"namespace":"test","name":"test"}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"handlers/nats.go:129","message":"connection status","context":{"status":1}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"handlers/nats.go:145","message":"successfully unsubscribed/deleted subscription","context":{}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"eventtype/clean.go:45","message":"cannot find application","context":{"name":"noapp"}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"eventtype/clean.go:53","message":"clean event-type","context":{"before":"sap.kyma.custom.noapp.order.created.v1","after":"sap.kyma.custom.noapp.order.created.v1"}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:221","message":"successfully created Nats subscriptions","context":{"kind":"Subscription","name":"test","namespace":"test","version":2}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:266","message":"successfully updated subscription status","context":{}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:135","message":"received subscription reconciliation request","context":{"namespace":"test","name":"test"}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"handlers/nats.go:129","message":"connection status","context":{"status":1}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"handlers/nats.go:145","message":"successfully unsubscribed/deleted subscription","context":{}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"eventtype/clean.go:45","message":"cannot find application","context":{"name":"noapp"}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"eventtype/clean.go:53","message":"clean event-type","context":{"before":"sap.kyma.custom.noapp.order.created.v1","after":"sap.kyma.custom.noapp.order.created.v1"}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:221","message":"successfully created Nats subscriptions","context":{"kind":"Subscription","name":"test","namespace":"test","version":2}}
{"level":"INFO","timestamp":"2022-01-04T08:34:02Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:266","message":"successfully updated subscription status","context":{}}


{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:135","message":"received subscription reconciliation request","context":{"namespace":"test","name":"test"}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"handlers/nats.go:129","message":"connection status","context":{"status":1}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"handlers/nats.go:145","message":"successfully unsubscribed/deleted subscription","context":{}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"eventtype/clean.go:45","message":"cannot find application","context":{"name":"noapp"}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"eventtype/clean.go:53","message":"clean event-type","context":{"before":"sap.kyma.custom.noapp.order.created.v1","after":"sap.kyma.custom.noapp.order.created.v1"}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:221","message":"successfully created Nats subscriptions","context":{"kind":"Subscription","name":"test","namespace":"test","version":2}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:266","message":"successfully updated subscription status","context":{}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:135","message":"received subscription reconciliation request","context":{"namespace":"test","name":"test"}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"handlers/nats.go:129","message":"connection status","context":{"status":1}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"handlers/nats.go:145","message":"successfully unsubscribed/deleted subscription","context":{}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"eventtype/clean.go:45","message":"cannot find application","context":{"name":"noapp"}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"eventtype/clean.go:53","message":"clean event-type","context":{"before":"sap.kyma.custom.noapp.order.created.v1","after":"sap.kyma.custom.noapp.order.created.v1"}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:221","message":"successfully created Nats subscriptions","context":{"kind":"Subscription","name":"test","namespace":"test","version":2}}
{"level":"INFO","timestamp":"2022-01-04T08:44:29Z","logger":"reconciler.Subscription","caller":"subscription-nats/reconciler.go:266","message":"successfully updated subscription status","context":{}}


```

**Comment by radufa** [comment]
The comment for 2.0.0 branch are valid here too:
https://github.com/kyma-project/kyma/issues/12978#issuecomment-1005637911
https://github.com/kyma-project/kyma/issues/12978#issuecomment-1007197751

**Comment by marcobebway** [comment]
Should be fixed for Kyma 2.0.
